library/alloc/src/string.rs

```diff
@@ -1940,8 +1940,10 @@ impl String {
 
     /// Converts this `String` into a <code>[Box]<[str]></code>.
     ///
-    /// This will drop any excess capacity.
+    /// Before doing the conversion, this method discards excess capacity like [`shrink_to_fit`].
+    /// Note that the [`shrink_to_fit`] call may reallocate and copy the bytes of the string.
```

Comment:
It feels odd to describe a side effect of a `shrink_to_fit` call, considering that `into_boxed_str` isn't previously described as calling `shrink_to_fit` (only that it discards excess capacity, like `shrink_to_fit`).

Reply:
Should probably convey the core point without mentioning implementation details like `shrink_to_fit`.

Reply:
Note that this phrasing was taken from `Vec::into_boxed_slice`, so if this should be changed, it should be changed there as well.

Reply:
> Note that this phrasing was taken from `Vec::into_boxed_slice`

That's the first sentence. My comment was about the second one (GitHub's UI does not make this clear, unfortunately).

Reply:
That's fair — changed the side-effect writing to not talk about `shrink_to_fit` :+1: